include/boost/http_proto/detail/filter.hpp

```diff
@@ -21,14 +20,16 @@
 
 namespace boost {
 namespace http_proto {
+namespace detail {
 
 /** A filter for buffers
 */
-class BOOST_HTTP_PROTO_DECL
-    filter
-    : public buffered_base
+class filter
```

Comment:
this type is still publicly visible. can't we just forward-declare it and only declare it in the .cpp files?

Reply:
lets do it